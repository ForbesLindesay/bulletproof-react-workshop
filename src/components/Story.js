// @flow

import React, {Component} from 'react';

type Props = {|
  +id: string,
  +votes: number,
  +body: string,
  +onVote: (id: string) => mixed,
|};

function Story(props: Props) {
  return (
    <article data-test-id="story">
      <span className="vote-count" data-test-id="VoteCount">
        {props.votes}
      </span>
      <div className="story-body" data-test-id="StoryBody">{props.body}</div>
      <button
        type="button"
        onClick={() => props.onVote(props.id)}
        data-test-id="AddVote"
      >
        Vote
      </button>
    </article>
  );
}

export default Story;
