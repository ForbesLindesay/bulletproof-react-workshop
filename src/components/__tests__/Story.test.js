// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import Story from '../Story';

test('Spinner', () => {
  expect(
    renderer
      .create(
        <Story votes={5} body="A great story" id="akjd" onVote={() => {}} />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
