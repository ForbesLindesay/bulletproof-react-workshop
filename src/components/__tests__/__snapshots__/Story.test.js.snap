// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner 1`] = `
<article
  data-test-id="story"
>
  <span
    className="vote-count"
    data-test-id="VoteCount"
  >
    5
  </span>
  <div
    className="story-body"
    data-test-id="StoryBody"
  >
    A great story
  </div>
  <button
    data-test-id="AddVote"
    onClick={[Function]}
    type="button"
  >
    Vote
  </button>
</article>
`;
